engine/CMakeLists.txt

```diff
@@ -651,21 +651,27 @@ ELSE (ENABLE_PIE)
 ENDIF (ENABLE_PIE)
 
 IF (MSVC)
-    SET(COMPILER_SPECIFIC_FLAGS "/FI\"${Vega_Strike_BINARY_DIR}\"/config.h  -D__x86_64__ /Z7")
-    set(TOOLCHAIN_SPECIFIC_LINK_FLAGS "/DEBUG:FULL")
+    SET(COMPILER_SPECIFIC_FLAGS "/FI\"${Vega_Strike_BINARY_DIR}\"/config.h  -D__x86_64__ ")
+    SET(COMPILER_SPECIFIC_DEBUG_FLAGS "/Z7 ") # also used for release-with-debug -- can change later
+    SET(COMPILER_SPECIFIC_PROFILER_FLAGS "/Z7 /GL ") # not tested -- requires additional linker flag specialization for profiling run ( /FASTGENPROFILE) and profile use ( /USEPROFILE)
+    SET(COMPILER_SPECIFIC_RELEASE_FLAGS " ")
+    SET(TOOLCHAIN_SPECIFIC_LINK_FLAGS "/DEBUG:FULL ")
 ELSE (MSVC)
-    SET(COMPILER_SPECIFIC_FLAGS "-include config.h -pipe -Og -g3 -Wall -std=c++11 -fvisibility=hidden")
-    set(TOOLCHAIN_SPECIFIC_LINK_FLAGS "-Bsymbolic ")
+    SET(COMPILER_SPECIFIC_FLAGS "-include config.h -pipe -Og -g3 -Wall -std=c++11 -fvisibility=hidden ")
```

Comment:
Still need to take out the `-Og -g3` from this line

Reply:
DOH.

Reply:
(Waiting for convergence on the other outstanding strncpy_s discussion to fix this with a single commit)